modin/pandas/test/test_dataframe.py

```diff
@@ -3003,316 +3001,136 @@ def test_hasattr_sparse(self, data):
             assert modin_result == pandas_result
 
 
-class TestDataFrameReduction_A:
-    @pytest.mark.parametrize("data", test_data_values, ids=test_data_keys)
-    @pytest.mark.parametrize("axis", axis_values, ids=axis_keys)
+class TestDataFrameReduction:
+    @pytest.mark.parametrize("method", ["all", "any"])
+    @pytest.mark.parametrize("is_transposed", [False, True])
     @pytest.mark.parametrize(
         "skipna", bool_arg_values, ids=arg_keys("skipna", bool_arg_keys)
     )
+    @pytest.mark.parametrize("axis", axis_values, ids=axis_keys)
+    @pytest.mark.parametrize("data", [test_data["dense_nan_data"]])
+    def test_all_any(self, data, axis, skipna, is_transposed, method):
+        eval_general(
+            *create_test_dfs(data),
+            lambda df: getattr((df.T if is_transposed else df), method)(
+                axis=axis, skipna=skipna, bool_only=None
+            ),
+        )
+
+    @pytest.mark.parametrize("method", ["all", "any"])
     @pytest.mark.parametrize(
         "bool_only", bool_arg_values, ids=arg_keys("bool_only", bool_arg_keys)
     )
-    def test_all(self, data, axis, skipna, bool_only):
-        modin_df = pd.DataFrame(data)
-        pandas_df = pandas.DataFrame(data)
-
-        try:
-            pandas_result = pandas_df.all(axis=axis, skipna=skipna, bool_only=bool_only)
-        except Exception as e:
-            with pytest.raises(type(e)):
-                modin_df.all(axis=axis, skipna=skipna, bool_only=bool_only)
-        else:
-            modin_result = modin_df.all(axis=axis, skipna=skipna, bool_only=bool_only)
-            df_equals(modin_result, pandas_result)
-
-        # Test when axis is None. This will get repeated but easier than using list in parameterize decorator
-        try:
-            pandas_result = pandas_df.all(axis=None, skipna=skipna, bool_only=bool_only)
-        except Exception as e:
-            with pytest.raises(type(e)):
-                modin_df.all(axis=None, skipna=skipna, bool_only=bool_only)
-        else:
-            modin_result = modin_df.all(axis=None, skipna=skipna, bool_only=bool_only)
-            df_equals(modin_result, pandas_result)
+    def test_all_any_specific(self, bool_only, method):
+        data = {
+            "float_col": [np.NaN, 9.4, 10.1, np.NaN],
+            "str_col": ["a", np.NaN, "c", "d"],
+            "bool_col": [False, True, True, False],
+        }
```

Comment:
it seems that it is a commonly used data set (repeats 5 times in your RP), maybe we should move it to the `test/utils.py`?

Reply:
will do